### PR TITLE
[FIX] account: avoid generating duplicate journal entry numbers

### PR DESCRIPTION
**Pull request by Florimond**
In Accounting:
- Go to 'Adviser > Journal Entries'
- Create a journal entry for 'Miscellaneous Operations'.
- Go to 'Configuration > Journals' and select 'Miscellaneous Operations'.
- Edit the journal, and set the 'Next Number' to 1.
- Go back to the Journal Entries and create another entry for same journal.

Before this commit:

A new entry will be generated with the same number as the previous one.

After this commit:

Error : 'Posted journal entry must have a unique sequence number per company.'

Backport of the constraint in commit beaa30a3d1843de43a45f419bfbc1bfa7613a920
which prevent creating two account moves with the same number. However, this
commit's constraint expect to check the current move which is in draft mode
against all other moves in database.
Also note that the field Type doesn't exist in versions < 13.

OPW: 2060313

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Florimond** [comment]
@smetl @qdp-odoo

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -20,6 +21,31 @@ class AccountMove(models.Model):
     _description = "Account Entry"
     _order = 'date desc, id desc'
 
+    @api.multi
+    @api.constrains('name', 'journal_id')
+    def _check_unique_sequence_number(self):
```
Comment (smetl) [inline addons/account/models/account_move.py]: Use the fixed version of the constraint:
https://github.com/odoo/odoo/blob/saas-12.4/addons/account/models/account_move.py#L1250

Comment (Florimond) [reply addons/account/models/account_move.py]: I've just updated the commit.

**Comment by Florimond** [comment]
These modification are cancelled.

For information, this constraint and the corresponding test are both present in 12.4:

Constraint in 12.4
https://github.com/odoo/odoo/blob/54049c14b3c3dc46992461a69b1d68fe9e665b14/addons/account/models/account_move.py#L1250

Test in 12.4
https://github.com/odoo/odoo/blob/54049c14b3c3dc46992461a69b1d68fe9e665b14/addons/account/tests/test_account_move_entry.py#L111

**Pull request closed without merging**